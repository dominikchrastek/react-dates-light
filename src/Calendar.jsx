/* @flow */
import * as React from "react";
import * as R from "ramda";
import styled from "styled-components";

import eachDayOfInterval from "date-fns/eachDayOfInterval";
import isSameDay from "date-fns/isSameDay";
import isBefore from "date-fns/isBefore";
import startOfMonth from "date-fns/startOfMonth";
import endOfMonth from "date-fns/endOfMonth";

import CalendarMonth from "./CalendarMonth";
import ArrowLeft from "./ArrowLeft";
import ArrowRight from "./ArrowRight";
import type { CalendarState, CalendarProps } from ".";
import * as utils from "./utils";

const defaultColors = {
  selected: "rgb(244, 114, 49)",
  selectedHover: "rgb(255, 141, 74)",
  border: "#e4e7e7",
  background: "white",
  hover: "#e4e7e7",
  disabled: "gray"
};

const StyledArrowLeft = styled(ArrowLeft)`
  position: absolute;
  top: 5px;
  left: 12.5px;
  fill: #82888a;
  width: 18px;
`;

const StyledArrowRight = styled(ArrowRight)`
  position: absolute;
  top: 5px;
  right: 12.5px;
  fill: #82888a;
  width: 18px;
`;

export const getWidth = (visibleMonths: number, width: number): string =>
  `${width * visibleMonths}px`; // 321 - 20 = 301 : 20 = margin

const CalendarWrapper = styled.div`
  position: relative;
  margin: 0 auto;
  text-align: center;
  max-width: ${props => getWidth(props.visibleMonths, props.width)};
`;

const CalendarMonthWrapper = styled.div`
  display: flex;
`;

const NavBtn = styled.button`
  border: 1px solid ${props => props.colors.border};
  position: absolute;
  background: ${props => props.colors.background};
  border-radius: 2px;
  width: 43px;
  height: 30px;
  cursor: pointer;
  :hover {
    background: ${props => props.colors.border};
  }
  &:focus {
    outline: none;
  }
  :disabled {
    cursor: not-allowed;
    :hover {
      background: initial;
    }
  }
`;

const StyledMonth = styled(CalendarMonth)`
  margin-right: 10px;
  &:last-child {
    margin-right: 0;
  }
`;

const PrevBtn = styled(NavBtn)`
  left: 0;
`;

const NextBtn = styled(NavBtn)`
  right: 0;
`;

type Props = CalendarProps;

type State = CalendarState;

class Calendar extends React.PureComponent<Props, State> {
  static defaultProps = {
    disabledDates: [],
    allowedDates: [],
    visibleMonths: 1,
    colors: {},
    className: "",
    classes: {},
    future: true,
    past: true,
    rangeSelect: false,
    showMonthName: true,
    showWeekDayNames: true,
    customClasses: {},
    weekDayFormat: "E",
    width: 301,
    firstWeekDay: 0
  };

  constructor(props: Props) {
    super(props);

    this.state = {
      end: null,
      hoveredDates: [],
      currentMonth: utils.getCurrentMonthIndex(
        props.firstMonth,
        props.lastMonth,
        props.selectedDates,
        props.future,
        props.visibleMonths
      ),
      isFocused: false,
      start: null,
      // eslint-disable-next-line react/no-unused-state
      selectedInternally: false
    };
  }

  componentWillUpdate(nextProps: Props, nextState: State) {
    const { selectedDates } = this.props;
    // when selectedDates came as a props, we need to know if they were changed
    // if they were, then we have to determine if they were changed internally
    // or externaly (from parent component)
    if (nextProps.selectedDates !== selectedDates) {
      this.handleSetCurrentMonth(nextProps, nextState);
    }
  }

  handleNext = () => {
    this.setState(state => ({
      currentMonth: state.currentMonth + 1
    }));
  };

  handlePrev = () => {
    this.setState(state => ({
      currentMonth: state.currentMonth - 1
    }));
  };

  /**
   * Method that will set current month
   * atm it's called only when dates are selected externally,
   * it means that dates were selected from parent component and pass as a
   * @argument {Props} nextProps
   * @argument {State} nextState
   */
  handleSetCurrentMonth = (nextProps: Props, nextState: State) => {
    const { firstMonth, lastMonth, future, visibleMonths } = this.props;
    // if date wasn't selected internally (it means that selectedDates
    // was changed from parent component ) then calculate current month and set it
    // also set selectedInternally to false
    if (!nextState.selectedInternally) {
      const currentMonth = utils.getCurrentMonthIndex(
        firstMonth,
        lastMonth,
        nextProps.selectedDates,
        future,
        visibleMonths
      );
      // eslint-disable-next-line react/no-unused-state
      this.setState({ selectedInternally: false, currentMonth });
      // }
      // otherwise just set selectedInternally to false, so we can determine if next
      // date select will be done in this component or in parent component
    } else {
      // eslint-disable-next-line react/no-unused-state
      this.setState({ selectedInternally: false });
    }
  };

  /**
   * Method for selecting date range
   * @argument {Date} date
   */
  handleSetRange = (date: Date) => {
    const { start } = this.state;
    const { disabledDates, selectDates } = this.props;
    // when someting is already selected (start)
    // and we clicked on some date (date)
    // then we will set array of dates between
    // date and start - it depends which one is first
    // selected array of dates is sorted ascending
    if (date && start) {
      // eslint-disable-next-line react/no-unused-state
      this.setState({ selectedInternally: true });
      if (isBefore(date, start)) {
        const range = eachDayOfInterval({ start: date, end: start });
        selectDates(R.without(disabledDates, range));
      } else {
        const range = eachDayOfInterval({ start, end: date });
        selectDates(R.without(disabledDates, range));
      }
    }
  };

  /**
   * Method for handling date select
   * @argument {Date} date
   */
  handleSelect = (date: Date) => {
    const { disabledDates, rangeSelect, selectDates } = this.props;
    const { isFocused, end, start } = this.state;
    // when something is already selected
    if (isFocused && rangeSelect) {
      this.setState({
        end: date,
        isFocused: false,
        hoveredDates: [],
        // eslint-disable-next-line react/no-unused-state
        selectedInternally: true
      });
      this.handleSetRange(date);
      // when unselecting date
    } else if (
      // $FlowExpected
      isSameDay(date, end) &&
      // $FlowExpected
      isSameDay(end, start)
    ) {
      this.setState({
        start: null,
        end: null,
        isFocused: false,
        // eslint-disable-next-line react/no-unused-state
        selectedInternally: true
      });
      selectDates([]);
      // initial select case
    } else {
      this.setState({
        start: date,
        isFocused: true,
        // eslint-disable-next-line react/no-unused-state
        selectedInternally: true
      });
      selectDates(R.without(disabledDates, [date]));
    }
  };

  /**
   * Method for handling date hover
   * @argument {Date} date
   */
  handleHover = (date: Date) => {
    const { rangeSelect } = this.props;
    const { isFocused, start } = this.state;
    if (start && date && isFocused && rangeSelect) {
      if (isBefore(date, start)) {
        this.setState({
          hoveredDates: eachDayOfInterval({ start: date, end: start })
        });
      } else {
        this.setState({
          hoveredDates: eachDayOfInterval({
            start,
            end: date
          })
        });
      }
    }
  };

  render() {
    const {
      visibleMonths,
      selectedDates,
      disabledDates,
      allowedDates,
      colors,
      className,
      classes,
      past,
      future,
      firstMonth,
      lastMonth,
      showMonthName,
      showWeekDayNames,
      customClasses,
      weekDayFormat,
      weekDayFormatter,
      monthNameFormatter,
      CustomTd,
      width,
      buttonBack,
      buttonForward,
      firstWeekDay
    } = this.props;

    const { currentMonth, hoveredDates, isFocused } = this.state;

    const mergedColors = R.merge(defaultColors, colors);
    const months = utils.getMonths(firstMonth, lastMonth);

    return (
      <CalendarWrapper
        className={className}
        visibleMonths={visibleMonths}
        width={width}
      >
        {months.length > visibleMonths && (
          <PrevBtn
            data-test="rdl-prev-button"
            className={classes.button || ""}
            onClick={this.handlePrev}
            disabled={currentMonth === 0 || months.length <= visibleMonths}
            colors={mergedColors}
          >
            {buttonBack ? (
              React.cloneElement(buttonBack, {
                disabled: currentMonth === 0 || months.length <= visibleMonths
              })
            ) : (
              <StyledArrowLeft />
            )}
          </PrevBtn>
        )}

        {months.length > visibleMonths && (
          <NextBtn
            data-test="rdl-next-button"
            className={classes.button || ""}
            onClick={this.handleNext}
            disabled={
              currentMonth === R.subtract(R.length(months), visibleMonths) ||
              months.length <= visibleMonths
            }
            colors={mergedColors}
          >
            {buttonForward ? (
              React.cloneElement(buttonForward, {
                disabled:
                  currentMonth ===
                    R.subtract(R.length(months), visibleMonths) ||
                  months.length <= visibleMonths
              })
            ) : (
              <StyledArrowRight />
            )}
          </NextBtn>
        )}

        <CalendarMonthWrapper className={classes.calendarWrapper || ""}>
          {R.map(
            month => (
              <StyledMonth
                key={month}
                month={month}
                CustomTd={CustomTd}
                selectedDates={selectedDates}
                disabledDates={disabledDates}
                allowedDates={allowedDates}
                selectDate={this.handleSelect}
                onHover={this.handleHover}
                hoveredDates={hoveredDates}
                past={past}
                isFocused={isFocused}
                future={future}
                colors={mergedColors}
                classes={classes}
                showMonthName={showMonthName}
                showWeekDayNames={showWeekDayNames}
                weekDayFormat={weekDayFormat}
                weekDayFormatter={weekDayFormatter}
                monthNameFormatter={monthNameFormatter}
                customClasses={utils.filterCustomClasses(
                  startOfMonth(month),
                  endOfMonth(month)
                )(customClasses)}
                width={width}
                firstWeekDay={firstWeekDay}
              />
            ),
            utils.calendarMonthsToRender(visibleMonths, currentMonth, months)
          )}
        </CalendarMonthWrapper>
      </CalendarWrapper>
    );
  }
}

export default Calendar;
