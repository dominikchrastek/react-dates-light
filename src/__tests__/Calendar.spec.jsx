import React from "react";
import { shallow, mount } from "enzyme";
import MockDate from "mockdate";

import * as utils from "../utils";

import Calendar, { getWidth } from "../Calendar";

describe("#Calendar", () => {
  const yesterday = new Date(2018, 2, 9);
  const today = new Date(2018, 2, 10);
  const tomorrow = new Date(2018, 2, 11);
  // const month = new Date(2018, 2, 2);
  const prevMonth = new Date(2018, 1, 1);
  const nextMonth = new Date(2018, 3, 3);
  beforeEach(() => {
    MockDate.set("2/31/2018");
  });
  afterEach(() => {
    MockDate.reset();
  });
  describe("methods", () => {
    it("should render correctly", () => {
      const wrapper = shallow(
        <Calendar
          className="wrapper"
          visibleMonths={3}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={jest.fn()}
          future={false}
          rangeSelect
        />
      );
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it("should render correctly with future enabled", () => {
      const wrapper = shallow(
        <Calendar
          className="wrapper"
          visibleMonths={3}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={jest.fn()}
          future
          rangeSelect
        />
      );
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it("should render correctly with allowedDates", () => {
      const wrapper = shallow(
        <Calendar
          className="wrapper"
          visibleMonths={3}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          allowedDates={[today]}
          selectDates={jest.fn()}
          future
          rangeSelect
        />
      );
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it("should render correctly with custom icons", () => {
      const wrapper = shallow(
        <Calendar
          className="wrapper"
          visibleMonths={3}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={jest.fn()}
          future={false}
          rangeSelect
          buttonLeft={<div>back</div>}
          buttonForward={<div>forward</div>}
        />
      );
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it("rdl-prev-button should works", () => {
      const wrapper = mount(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={jest.fn()}
          rangeSelect
        />
      );

      const node = wrapper.instance();
      expect(node.state.currentMonth).toBe(1);
      wrapper
        .find('[data-test="rdl-next-button"]')
        .at(1)
        .simulate("click");

      expect(node.state.currentMonth).toBe(2);
      wrapper
        .find('[data-test="rdl-next-button"]')
        .at(1)
        .simulate("click");
      expect(
        wrapper
          .find('[data-test="rdl-next-button"]')
          .at(1)
          .is("[disabled]")
      ).toBeTruthy();
      expect(node.state.currentMonth).toBe(2);
    });

    it("rdl-prev-button should works", () => {
      const wrapper = mount(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={jest.fn()}
          rangeSelect
        />
      );

      const node = wrapper.instance();

      expect(node.state.currentMonth).toBe(1);
      wrapper
        .find('[data-test="rdl-prev-button"]')
        .at(1)
        .simulate("click");

      expect(node.state.currentMonth).toBe(0);
      wrapper
        .find('[data-test="rdl-prev-button"]')
        .at(1)
        .simulate("click");
      expect(
        wrapper
          .find('[data-test="rdl-prev-button"]')
          .at(1)
          .is("[disabled]")
      ).toBeTruthy();
      expect(node.state.currentMonth).toBe(0);
    });

    it("handleSelect select range: prev date - next date", () => {
      const selectDates = jest.fn();
      const wrapper = shallow(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={selectDates}
          rangeSelect
        />
      );

      const node = wrapper.instance();

      node.handleSelect(yesterday);

      expect(node.state.start).toBe(yesterday);
      expect(node.state.isFocused).toBe(true);
      expect(node.state.selectedInternally).toBe(true);
      // call selectDates with selected dates
      expect(selectDates).toBeCalledWith([yesterday]);

      node.handleSelect(today);
      expect(node.state.end).toBe(today);
      expect(node.state.isFocused).toBe(false);
      expect(node.state.hoveredDates).toEqual([]);
      expect(node.state.selectedInternally).toBe(true);

      // handleSetSelected should be called
      expect(node.state.selectedInternally).toBe(true);
      expect(selectDates).toBeCalledWith([yesterday, today]);
    });

    it("handleSelect select range: next date - prev date", () => {
      const selectDates = jest.fn();
      const wrapper = shallow(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={selectDates}
          rangeSelect
        />
      );

      const node = wrapper.instance();

      node.handleSelect(today);
      expect(node.state.start).toBe(today);
      expect(node.state.isFocused).toBe(true);
      expect(node.state.hoveredDates).toEqual([]);
      expect(node.state.selectedInternally).toBe(true);
      // call selectDates with selected dates
      expect(selectDates).toBeCalledWith([today]);

      node.handleSelect(yesterday);
      expect(node.state.end).toBe(yesterday);
      expect(node.state.isFocused).toBe(false);
      expect(node.state.selectedInternally).toBe(true);

      // handleSetSelected should be called
      expect(node.state.selectedInternally).toBe(true);
      expect(selectDates).toBeCalledWith([yesterday, today]);
    });
    it("handleSelect select same day and then unselect", () => {
      const selectDates = jest.fn();
      const wrapper = mount(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={selectDates}
          rangeSelect
        />
      );

      const node = wrapper.instance();

      node.handleSelect(today);

      expect(node.state.start).toBe(today);
      expect(node.state.isFocused).toBe(true);
      expect(node.state.selectedInternally).toBe(true);
      // call selectDates with selected dates
      expect(selectDates).toBeCalledWith([today]);

      node.handleSelect(today);
      expect(node.state.end).toBe(today);
      expect(node.state.isFocused).toBe(false);
      expect(node.state.selectedInternally).toBe(true);

      // handleSetSelected should be called
      expect(node.state.selectedInternally).toBe(true);
      expect(selectDates).toBeCalledWith([today]);

      // unselect
      node.handleSelect(today);
      expect(node.state.start).toBe(null);
      expect(node.state.end).toBe(null);
      expect(node.state.isFocused).toBe(false);
      expect(node.state.selectedInternally).toBe(true);
      expect(selectDates).toBeCalledWith([]);
    });
    it("handleSetRange - nothing should happend", () => {
      const selectDates = jest.fn();
      const wrapper = mount(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={selectDates}
          rangeSelect
        />
      );

      const node = wrapper.instance();

      node.handleSetRange(today);
      expect(selectDates).not.toBeCalled();
    });

    it("handleHover - select date and hover it", () => {
      const selectDates = jest.fn();
      const wrapper = mount(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={selectDates}
          rangeSelect
        />
      );

      const node = wrapper.instance();
      node.handleSelect(today);

      expect(node.state.start).toBe(today);
      expect(node.state.isFocused).toBe(true);
      expect(node.state.selectedInternally).toBe(true);
      // call selectDates with selected dates
      expect(selectDates).toBeCalledWith([today]);

      node.handleHover(today);
      expect(node.state.hoveredDates).toEqual([today]);
    });

    it("handleHover - select date and hover next date", () => {
      const selectDates = jest.fn();
      const wrapper = mount(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={selectDates}
          rangeSelect
        />
      );

      const node = wrapper.instance();
      node.handleSelect(today);

      expect(node.state.start).toBe(today);
      expect(node.state.isFocused).toBe(true);
      expect(node.state.selectedInternally).toBe(true);
      // call selectDates with selected dates
      expect(selectDates).toBeCalledWith([today]);

      node.handleHover(tomorrow);
      expect(node.state.hoveredDates).toEqual([today, tomorrow]);
    });

    it("handleHover - select date and hover prev date", () => {
      const selectDates = jest.fn();
      const wrapper = mount(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={selectDates}
          rangeSelect
        />
      );

      const node = wrapper.instance();
      node.handleSelect(today);

      expect(node.state.start).toBe(today);
      expect(node.state.isFocused).toBe(true);
      expect(node.state.selectedInternally).toBe(true);
      // call selectDates with selected dates
      expect(selectDates).toBeCalledWith([today]);

      node.handleHover(yesterday);
      expect(node.state.hoveredDates).toEqual([yesterday, today]);
    });

    it("handleHover - nothing should happend", () => {
      const selectDates = jest.fn();
      const wrapper = mount(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={selectDates}
          rangeSelect
        />
      );

      const node = wrapper.instance();

      node.handleHover(today);
      expect(node.state.hoveredDates).toEqual([]);
    });

    it("handleSetCurrentMonth - componentWillUpdate, selected externally", () => {
      const nextProps = {
        selectedDates: today
      };
      const nextState = {
        selectedInternally: false
      };
      const selectDates = jest.fn();
      const wrapper = mount(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={selectDates}
          rangeSelect
        />
      );

      const node = wrapper.instance();
      node.componentWillUpdate(nextProps, nextState);

      // handleSetCurrentMonth should be called
      const currentMonth = utils.getCurrentMonthIndex(
        node.props.firstMonth,
        node.props.lastMonth,
        nextProps.selectedDates,
        node.props.future,
        node.props.visibleMonths
      );

      expect(node.state.selectedInternally).toBe(false);
      expect(node.state.currentMonth).toBe(currentMonth);
    });
    it("handleSetCurrentMonth - componentWillUpdate, selected internally", () => {
      const nextProps = {
        selectedDates: today
      };
      const nextState = {
        selectedInternally: true
      };
      const selectDates = jest.fn();
      const wrapper = mount(
        <Calendar
          className="wrapper"
          visibleMonths={1}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={selectDates}
          rangeSelect
        />
      );

      const node = wrapper.instance();
      node.componentWillUpdate(nextProps, nextState);

      // handleSetCurrentMonth should be called
      const currentMonth = utils.getCurrentMonthIndex(
        node.props.firstMonth,
        node.props.lastMonth,
        nextProps.selectedDates,
        node.props.future,
        node.props.visibleMonths
      );

      expect(node.state.selectedInternally).toBe(false);
      expect(node.state.currentMonth).not.toBe(currentMonth);
    });

    it("should render without month and week days names", () => {
      const wrapper = shallow(
        <Calendar
          className="wrapper"
          visibleMonths={3}
          firstMonth={prevMonth}
          lastMonth={nextMonth}
          selectedDates={[]}
          selectDates={jest.fn()}
          future={false}
          rangeSelect
          showMonthName={false}
          showWeekDayNames={false}
        />
      );
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });

  describe("helpers", () => {
    it("getWidth", () => {
      expect(getWidth(1, 301)).toBe("301px");
      expect(getWidth(3, 301)).toBe(`903px`);
      expect(getWidth(5, 200)).toBe(`1000px`);
    });
  });
});
