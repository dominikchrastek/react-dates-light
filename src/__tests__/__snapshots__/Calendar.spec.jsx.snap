// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Calendar methods should render correctly 1`] = `
<styled.div
  className="wrapper"
  visibleMonths={3}
  width={301}
>
  <styled.div
    className=""
  >
    <Styled(CalendarMonth)
      allowedDates={Array []}
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={false}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-02-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={true}
      showWeekDayNames={true}
      weekDayFormat="E"
      width={301}
    />
    <Styled(CalendarMonth)
      allowedDates={Array []}
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={false}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-03-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={true}
      showWeekDayNames={true}
      weekDayFormat="E"
      width={301}
    />
    <Styled(CalendarMonth)
      allowedDates={Array []}
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={false}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-04-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={true}
      showWeekDayNames={true}
      weekDayFormat="E"
      width={301}
    />
  </styled.div>
</styled.div>
`;

exports[`#Calendar methods should render correctly with allowedDates 1`] = `
<styled.div
  className="wrapper"
  visibleMonths={3}
  width={301}
>
  <styled.div
    className=""
  >
    <Styled(CalendarMonth)
      allowedDates={
        Array [
          2018-03-10T00:00:00.000Z,
        ]
      }
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={true}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-02-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={true}
      showWeekDayNames={true}
      weekDayFormat="E"
      width={301}
    />
    <Styled(CalendarMonth)
      allowedDates={
        Array [
          2018-03-10T00:00:00.000Z,
        ]
      }
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={true}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-03-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={true}
      showWeekDayNames={true}
      weekDayFormat="E"
      width={301}
    />
    <Styled(CalendarMonth)
      allowedDates={
        Array [
          2018-03-10T00:00:00.000Z,
        ]
      }
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={true}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-04-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={true}
      showWeekDayNames={true}
      weekDayFormat="E"
      width={301}
    />
  </styled.div>
</styled.div>
`;

exports[`#Calendar methods should render correctly with future enabled 1`] = `
<styled.div
  className="wrapper"
  visibleMonths={3}
  width={301}
>
  <styled.div
    className=""
  >
    <Styled(CalendarMonth)
      allowedDates={Array []}
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={true}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-02-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={true}
      showWeekDayNames={true}
      weekDayFormat="E"
      width={301}
    />
    <Styled(CalendarMonth)
      allowedDates={Array []}
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={true}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-03-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={true}
      showWeekDayNames={true}
      weekDayFormat="E"
      width={301}
    />
    <Styled(CalendarMonth)
      allowedDates={Array []}
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={true}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-04-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={true}
      showWeekDayNames={true}
      weekDayFormat="E"
      width={301}
    />
  </styled.div>
</styled.div>
`;

exports[`#Calendar methods should render without month and week days names 1`] = `
<styled.div
  className="wrapper"
  visibleMonths={3}
  width={301}
>
  <styled.div
    className=""
  >
    <Styled(CalendarMonth)
      allowedDates={Array []}
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={false}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-02-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={false}
      showWeekDayNames={false}
      weekDayFormat="E"
      width={301}
    />
    <Styled(CalendarMonth)
      allowedDates={Array []}
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={false}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-03-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={false}
      showWeekDayNames={false}
      weekDayFormat="E"
      width={301}
    />
    <Styled(CalendarMonth)
      allowedDates={Array []}
      classes={Object {}}
      colors={
        Object {
          "background": "white",
          "border": "#e4e7e7",
          "disabled": "gray",
          "hover": "#e4e7e7",
          "selected": "rgb(244, 114, 49)",
          "selectedHover": "rgb(255, 141, 74)",
        }
      }
      customClasses={Object {}}
      disabledDates={Array []}
      future={false}
      hoveredDates={Array []}
      isFocused={false}
      month={2018-04-01T00:00:00.000Z}
      onHover={[Function]}
      past={true}
      selectDate={[Function]}
      selectedDates={Array []}
      showMonthName={false}
      showWeekDayNames={false}
      weekDayFormat="E"
      width={301}
    />
  </styled.div>
</styled.div>
`;
